src/s1reader/s1_orbit.py

```diff
@@ -56,16 +61,69 @@ def download_orbit(safe_file: str, orbit_dir: str):
     # Find precise orbit first
     orbit_dict = get_orbit_dict(mission_id, start_time,
                                 end_time, 'AUX_POEORB')
+
     # If orbit dict is empty, find restituted orbits
     if orbit_dict is None:
         orbit_dict = get_orbit_dict(mission_id, start_time,
                                     end_time, 'AUX_RESORB')
+
     # Download orbit file
-    orbit_file = os.path.join(orbit_dir, f"{orbit_dict['orbit_name']}.EOF")
-    if not os.path.exists(orbit_file):
-        download_orbit_file(orbit_dir, orbit_dict['orbit_url'])
+    if orbit_dict is not None:
+        orbit_file = os.path.join(orbit_dir, f"{orbit_dict['orbit_name']}.EOF")
+        if not os.path.exists(orbit_file):
+            download_orbit_file(orbit_dir, orbit_dict['orbit_url'])
 
-    return orbit_file
+        return orbit_file
+
+    # POEORB is not found, or there is no RESORB file that
+    # covers the sensing period + margin at the starting time.
+    # Try to find two subsequent RESORB files that covers the
+    # sensing period + margins at the starting time.
+    if orbit_dict is None:
```

Comment:
This if-statement here looks useless to me. I think at this point `orbit_dict` cannot be `not None`. Otherwise, it would have entered the if-statement in line `71` and escaped in line `76`.

Reply:
makes sense. I've removed the it statement here.